Babel

surma opened the pull request:
r: @gauntface

surma commented on test/data/valid-files/es2015-module-multiple-exports.js:
```diff
@@ -0,0 +1,3 @@
+export const plusOne = x => x + 1;
+export function sum(x = 0, y = 0, z = 0) { return x + y + z; }
+export default "default";
```
Should we do subfolders in `valid-files` per task?

surma commented on test/task-checker.js:
```diff
@@ -46,51 +48,53 @@ describe('This should enforce any rules we want to exist for our tasks', () => {
-
-        done();
+      it('should build cleanly', function(done) {
+        this.timeout(10000);
```
Yeah. This is intentional. Babel is friggin slow, yo.

![screenshot 2016-01-20 13 44 14](https://cloud.githubusercontent.com/assets/234957/12450530/43429e4a-bf7c-11e5-811b-b78f0161e724.png)

surma replied:
We could also go for `this.timeout(0)` to disable timeouts all together.

gauntface replied:
Nah keep the timeout in place but 3 seconds for that single file is a little alarming tbh :-/

surma replied:
It might be that the transpilation itself is fast but Babel has a huge warmup time. We should keep in eye out.

gauntface commented on src/wsk-tasks/babel.js:
```diff
@@ -0,0 +1,42 @@
+var uglify = require('gulp-uglify');
+var sourcemaps = require('gulp-sourcemaps');
+
+function build() {
```
Does babel bundle imports?

I've only ever used browserify with babelify.

surma replied:
Nope. Forgot about that. WIll fix

surma replied:
What is your preferred approach to this? Should I just steal the browserify from app-shell?

gauntface replied:
I'm fine with the app-shell approach, although I'm sure it can be tidied up.

It was taken from aerotwist's voice memos I believe and just made to be generic.

ramasilveyra commented on test/task-checker.js:
```diff
@@ -24,8 +24,9 @@
 require('chai').should();
 
 var sass = require('../src/wsk-tasks/sass.js');
+var babel = require('../src/wsk-tasks/babel.js');
```
It would be better use `const` for require? e.g. `const babel = require('../src/wsk-tasks/babel.js')`

robwierzbowski replied:
👍🏽👍🏽👍🏽

gauntface replied:
Makes sense to me :)

surma replied:
Agreed.

sindresorhus commented on src/wsk-tasks/babel.js:
```diff
@@ -0,0 +1,42 @@
+  var stream = gulp.src(GLOBAL.config.src + '/**/*.js')
+    .pipe(sourcemaps.init())
+    .pipe(babel({
+      presets: ["es2015"]
```
single-quotes

surma replied:
My  bad.

sindresorhus commented on src/wsk-tasks/babel.js:
```diff
@@ -0,0 +1,42 @@
+    }));
+
+  if (GLOBAL.config.env === 'prod') {
+    stream = stream.pipe(uglify());
```
https://github.com/robrich/gulp-if would make this look better.

robwierzbowski replied:
Agree with Sindre about gulp-if.

On Tuesday, January 26, 2016, Sindre Sorhus <notifications@github.com>
wrote:

> In src/wsk-tasks/babel.js
> <https://github.com/google/web-starter-kit/pull/818#discussion_r50829741>:
>
> > + */
> > +
> > +var gulp = require('gulp');
> > +var babel = require('gulp-babel');
> > +var uglify = require('gulp-uglify');
> > +var sourcemaps = require('gulp-sourcemaps');
> > +
> > +function build() {
> > +  var stream = gulp.src(GLOBAL.config.src + '/**/*.js')
> > +    .pipe(sourcemaps.init())
> > +    .pipe(babel({
> > +      presets: ["es2015"]
> > +    }));
> > +
> > +  if (GLOBAL.config.env === 'prod') {
> > +    stream = stream.pipe(uglify());
>
> https://github.com/robrich/gulp-if would make this look better.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/web-starter-kit/pull/818/files#r50829741>.
>


-- 
Rob Wierzbowski
@robwierzbowski <http://twitter.com/#!/robwierzbowski>
http://github.com/robwierzbowski
http://robwierzbowski.com

surma replied:
This was actually a [conscious decision](https://github.com/google/web-starter-kit/pull/813#discussion_r49863203) by @gauntface and myself. I think that there’s very little benefit in `gulp-if` and actually makes things seem more magic (apart from the fact that the condition part can behave rather strange). In this case, I’d agree that it makes things more readable, but in general it can get messy, esp. if you want to run multiple plugins conditionally.

I don’t mind being out-voted on this one, though ;)

robwierzbowski replied:
`pumpify` works well with gulp-if, but I agree it can get less pretty if the conditionals are complex.

gauntface replied:
I am trying to cut down dependencies as much as possible and gulp-if is a nice to have not vital.

Other side of is that with gulp-if, it will actually execute the code passed as an argument, which for one of the modules in app-shell caused a crash.

Compare:

.pipe(gulpif(
        false,
        license(DoesntExistVar)
))

vs

if (false) {
        license(DoesntExistVar);
}

with gulpif, license will still be executed with the variable, the plain old if, it wont. Minor thing, but was a tad annoying.

Do people feel strongly that it should be used?

surma commented:
Added `browserify` and implemented the rest of the review. Haven’t really tested the task, would like to use this opportunity to add task-specific tests in a separate PR. @gauntface WDYT?

surma commented:
Hold on this. I am not thinking straight today. We already have tests and they are not passing :smile:

surma commented:
Alright, pivoted to feed into rollup then into babel. Seems to work. Review at will :)

gauntface commented:
Going to merge and then fix up travis
